libMBIN/Source/NMS/GameComponents/GcMessageCrime.cs

```diff
@@ -1,15 +1,23 @@
-﻿using libMBIN.NMS.Toolkit;
+﻿using System.Collections.Generic;
+
+using libMBIN.NMS.Toolkit;
 using libMBIN.NMS.GameComponents;
 
 namespace libMBIN.NMS.GameComponents
 {
-	[NMS(Size = 0x30, GUID = 0x1CCB7BDC1ECBB02A, NameHash = 0xC64AF7F76101CD0)]
+    [NMS(Size = 0x30, GUID = 0x1CCB7BDC1ECBB02A, NameHash = 0x0C64AF7F76101CD0)]
     public class GcMessageCrime : NMSTemplate
     {
-		public enum CrimeEnum { AttackCreature, AttackSentinel, KillCreature, KillSentinel, MineResources, HitResources, AttackSpaceStation, AttackShip, AttackPolice,
-            KillShip, KillPolice, TimedShootable }
-		public CrimeEnum Crime;
-        public Vector4f Position;
-        public int Value;
+        // 0xD entries
+        public enum CrimeEnum { AttackCreature, AttackSentinel, AttackSentinelLaser, KillCreature, KillSentinel, MineResources,
+            HitResources, AttackSpaceStation, AttackShip, AttackPolice, KillShip, KillPolice, TimedShootable
+        }
+		/* 0x00 */ public CrimeEnum Crime;
+        /* 0x10 */ public Vector3f Position;
+        [NMS(Size = 0x8, Ignore = true)]
+        /* 0x20 */ public byte[] Padding20;
```

Comment:
something about this is wrong... There shouldn't be 8 bytes of padding before an int like this? I haven't got IDA open to check but this seems weird...

Reply:
I know but it is what it is!

>                              LAB_14050a701                                   XREF[2]:     14050a6b9(j), 14050a6ed(j)  
    .text:14050a701 4c 8d 05        LEA        R8, [s_Position_1422f7b38]                       ; = "Position"
    .text:14050a708 49 8b d6        MOV        RDX, R14
    .text:14050a70b 48 8d 4e 10     LEA        RCX, [RSI + 0x10]
    .text:14050a70f e8 1c c0        CALL       Alt_Vector3f_0x10_XYZ_pre                        ; undefined Alt_Vector3f_0x10_XYZ_pre(void)
    .text:14050a714 4c 8d 05        LEA        R8, [s_Value_14233770c]                          ; = "Value"
    .text:14050a71b 49 8b d6        MOV        RDX, R14
    .text:14050a71e 48 8d 4e 28     LEA        RCX, [RSI + 0x28]
    .text:14050a722 4c 8d 9c        LEA        R11=>local_28, [RSP + 0xb0]
    .text:14050a72a 49 8b 5b 38     MOV        RBX, qword ptr [R11 + local_res10]
    .text:14050a72e 49 8b 6b 40     MOV        RBP, qword ptr [R11 + local_res18]
    .text:14050a732 49 8b e3        MOV        RSP, R11
    .text:14050a735 41 5f           POP        R15
    .text:14050a737 41 5e           POP        R14
    .text:14050a739 41 5c           POP        R12
    .text:14050a73b 5f              POP        RDI
    .text:14050a73c 5e              POP        RSI
    .text:14050a73d e9 1e b7        JMP        NMS_int_0x4                                      ; undefined NMS_int_0x4()
                             -- Flow Override: CALL_RETURN (CALL_TERMINATOR)
                             ********** GcMessageCrime Exit **********

Reply:
Hmmm... very strange...